### PR TITLE
Final PR: Phase4, Phase5

### PR DESCRIPTION
**Pull request by KOVALKO2**
Phase 4:

1. All selectors are moved to a PageObject classes to a "pages" package. Test methods/classes doesn't contain any selectors. - **POs are in "pages" folder; locators are in "elements" folder**
2. All UI element that requires a wait must contain it. Wait must be a part of function at the level of PageObject - **waits are included into methods of elements and POs, where waits relate not to particular element, but to the page**  
3. Wait must be added to the function which is going to interact with UI element before the code that intends interacts with element.
4. New pull request must be created. One of mentors must be assigned as a review for the PR.
5. Build must be successfully builded on CircleCi - **fought with selenium for stability for 2 weeks and this is the best results, I achieved(( if you'll find some obvious causes of tests instability, please tell me! Currently I'm thinking about parallelism influence and thread.local things...**

Phase 5:

1. Test case steps updated with @allure.step tags.
2. Build on CI (Artifacts tab) must contains Allure reports. **test-reports \ allure \ allure-report \ index.html**
3. Reports for UI tests must contains screenshots.
4. Build must be successfully builded on CircleCi.
5. Readme file with status badge and command showing how to run your test

### CODE REVIEW

File: project/tests/ui_tests/test_ui_issues.py [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+            logger.info("Step 'Filtering list of my issues to get no issues' started")
+            issues_page.filter_issues("UIbkkbUIbkkbUIbkkb", results_expected = False)
+            assertion("Empty results notification is shown", issues_page.elems.empty_results_txt.get_text(),
+                      is_("No issues were found to match your search"),
+                      "No notification text found or some issues are shown")
+            logger.info("Step 'Filtering list of my issues to get no issues' finished")
+        logger.info("Test 'Search UI issue' finished")
+
+
+    @allure.title("Update UI issue")
```
Comment (archick12) [inline project/tests/ui_tests/test_ui_issues.py]: Please, add UI mark

File: project/tests/ui_tests/test_ui_issues.py [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+    def test_update_issue(self, fixture_login, correct_creds, fixture_update_issues_ui, fixture_update_issue_params,
+                          fixture_provide_webdriver_logged_in):
+        """
+        Update UI issue
+        """
+        logger.info("Test 'Update UI issue' started")
+        with allure.step("Getting ID of the issue to be updated"):
+            logger.info("Step 'Getting ID of the issue to be updated' started")
+            self.API_HTTP.session.headers['Content-Type'] = 'application/json'
```
Comment (archick12) [inline project/tests/ui_tests/test_ui_issues.py]: Could you wrap this code using method with nice and meaningful name?

Comment (archick12) [reply project/tests/ui_tests/test_ui_issues.py]: self.API_HTTP.session.headers['Content-Type'] = 'application/json'
 response_search_issue = self.API_HTTP.search(
                    url_part = self.ROUTES['search']['path'],
                    jql = "project = {0} AND summary ~ 'UIu*' AND issuetype = Bug AND reporter in ({1})"
                        .format(read_ini_file('../tests_config.ini')['Jira']['project_key'], correct_creds['login']))
            attach_json(dict(response_search_issue.headers), "Search issue Response headers")
            attach_json(list(response_search_issue.cookies.items()), "Search issue Response cookies")
            verify_http_status_code(response_search_issue.status_code, 200)
            attach_json(json.loads(response_search_issue.content), "Search issue Response body")
            issue_id = json.loads(response_search_issue.content)['issues'][0]['id']

File: project/tests/ui_tests/test_ui_issues.py (hunk 1 above)
Comment (KOVALKO2) [reply project/tests/ui_tests/test_ui_issues.py]: It is added for parent class: project/tests/base_ui_test.py::base_ui_test.BaseUiTest

File: project/tests/ui_tests/test_ui_issues.py (hunk 2 above)
Comment (KOVALKO2) [reply project/tests/ui_tests/test_ui_issues.py]: extracted it to method project/tests/global_test.GlobalTest#get_id_of_issue_to_update and used it in both UI and API tests

**Pull request merged**